pkg/buildpipeline/pipeline_test.go

```diff
@@ -11,6 +11,10 @@ import (
 // TODO: Probably move the YAML to external files, like in Declarative's tests, and write a builder for generating the
 // expected objects. Because as this is now, there are way too many lines here.
 func TestParseJenkinsfileYaml(t *testing.T) {
+	// Needed to take address of strings since workspace is *string. Is there a better way to handle optional values?
+	defaultWorkspace := "default"
```

Comment:
Maybe better for `workspace` to be a plain `string`, not a pointer, and use `""` as the default value? I looked into setting default values in the yaml unmarshaller but couldn't get it working in a few minutes and moved on.